internal/app/components/courses/courseRecords_test.go

```diff
@@ -399,3 +400,110 @@ func TestCourseRecords_Exclude(t *testing.T) {
 		})
 	}
 }
+
+func TestCourseRecord_IsPrereqSatisfied(t *testing.T) {
+	currTime := time.Now()
+	type fields struct {
+		Course         Course
+		Id             CourseRecordId
+		Grade          CourseGrade
+		CompletionDate *time.Time
+		Override       bool
+	}
+	type args struct {
+		pastRecords *CourseRecords
+	}
+	tests := []struct {
+		name   string
+		fields fields
+		args   args
+		want   bool
+	}{
+		{
+			name: "satisfied",
+			fields: fields{
+				Course: Course{
+					Id: 0,
+					Prereqs: CourseRequirement{
+						CourseId: 1,
+						MinGrade: 60,
+					},
+				},
+			},
+			args: args{
+				pastRecords: &CourseRecords{
+					&CourseRecord{
+						Course: Course{
+							Id: 1,
+						},
+						Grade: 10, // should still be satisfied because it's a future record
```

Comment:
theoretically Grade field should be empty if it's a future record right

Reply:
yeah, the purpose of that is to test if any of our logic is dependent on grade. This is the correct expected behaviour